### PR TITLE
Document patching in overview

### PR DESCRIPTION
**Pull request by justinvdm**
Most endpoints are likely to only support `PUT`s initially instead of `PUT`s and `PATCH`es, but it would be helpful to have a design to follow once we do start implementing the `PATCH`es.

### CODE REVIEW

**Comment by justinvdm** [comment]
Ready for review.

File: docs/index.rst [hunk 1]
```diff
@@ -45,6 +45,16 @@ Unless otherwise specified, the following verbs are used for requests:
 | DELETE | Used to delete a resource                                  |
 +--------+------------------------------------------------------------+
 
+Partial updates
+~~~~~~~~~~~~~~~
+
+Partial update requests should provide instructions for how a resource should be modified according to the structure proposed in `RFC 6902`_. Note that partial updates are *atomic*, so if any of the given instructions fail, none of the instructures will be carried out.
+
+.. _RFC 6902: http://tools.ietf.org/html/rfc6902#section-4.6
```
Comment (hodgestar) [inline docs/index.rst]: Do we actually want the `#section-4.6` here?

**Comment by hodgestar** [comment]
One minor comment, otherwise looks good. Woot on finding these RFCs.

File: docs/index.rst (hunk 1 above)
Comment (justinvdm) [inline docs/index.rst]: Woops, thanks, fixed.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**